### ISSUE TITLE
Event definitions for AppSync Events

### ISSUE THREAD

**Issue by nicklaswj** [awslabs/aws-lambda-rust-runtime#940]
There are currently no event definitions for the relatively new AWS AppSync Event service. Going through the [AWS docs](https://docs.aws.amazon.com/appsync/latest/eventapi/event-api-welcome.html), I've collected the following list of events that could be added:

1. [onPublish](https://docs.aws.amazon.com/appsync/latest/eventapi/channel-namespace-handlers.html#event-handler-onpublish)
2. [onSubscribe](https://docs.aws.amazon.com/appsync/latest/eventapi/channel-namespace-handlers.html#event-handler-onpublish) 
3. [Lambda authorization ](https://docs.aws.amazon.com/appsync/latest/eventapi/configure-event-api-auth.html#aws-lambda-authorization), consisting of a `authorizationRequest`, `authorizationContext` and a `authorizationReponse`.
4. Optionally/debatable definitions for the [event API websocket protocol messages](https://docs.aws.amazon.com/appsync/latest/eventapi/event-api-websocket-protocol.html#realtime-websocket-operations).

I'm open to adding these myself, since I'm currently developing a system that will use AppSync Event, so I need to write them regardless.

Should these definitions live in `aws_lambda_events::events::appsync` (potentially in a sub-module) or in a new module `aws_lambda_events::events::appsync_events` ?

**Comment by th0per** [comment]
@nicklaswj any status update on this? I am in need of the Lambda authorization event.

**Comment by nicklaswj** [comment]
Yes and no. I did implement the definitions myself, but since there was no activity here I placed it in our private repository... I can ask if we can open source it, but I don't know how long the process will be.

**Comment by GradExMachina** [comment]
Hi, if you'd like to cut us a PR, I'd be happy to review :)
